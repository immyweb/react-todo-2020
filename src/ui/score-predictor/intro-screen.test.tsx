import * as React from 'react';
import { shallow } from 'enzyme';

import IntroScreen from './intro-screen';
import Button from '../buttons/buttons';

function shallowRender(props?: any) {
  return shallow(
    <IntroScreen
      slideWidth={400}
      onStartGame={() => {}}
      introHeadline={'Premier League Predictor'}
      introTxt={'Play our awesome game'}
      {...props}
    />,
  );
}

describe('<IntroScreen />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('onStartGame is called when button is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onStartGame: fn });
    const btn = wrapper.find(Button);
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
