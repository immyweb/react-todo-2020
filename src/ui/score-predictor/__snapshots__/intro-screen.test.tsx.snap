// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroScreen /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IntroScreen
    introHeadline="Premier League Predictor"
    introTxt="Play our awesome game"
    onStartGame={[Function]}
    slideWidth={400}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <glamorous(div)
        className="ttt-inner"
      >
        <glamorous(glamorous(h3))
          fontColour="#459977"
        >
          Premier League Predictor
        </glamorous(glamorous(h3))>
        <glamorous(p)>
          Play our awesome game
        </glamorous(p)>
        <glamorous(button)
          background="#459977"
          fullWidth={true}
          onClick={[Function]}
        >
          Play Now
        </glamorous(button)>
      </glamorous(div)>,
      "width": 400,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            Premier League Predictor
          </glamorous(glamorous(h3))>,
          <glamorous(p)>
            Play our awesome game
          </glamorous(p)>,
          <glamorous(button)
            background="#459977"
            fullWidth={true}
            onClick={[Function]}
          >
            Play Now
          </glamorous(button)>,
        ],
        "className": "ttt-inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Premier League Predictor",
            "fontColour": "#459977",
          },
          "ref": null,
          "rendered": "Premier League Predictor",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Play our awesome game",
          },
          "ref": null,
          "rendered": "Play our awesome game",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "background": "#459977",
            "children": "Play Now",
            "fullWidth": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Play Now",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(div)
          className="ttt-inner"
        >
          <glamorous(glamorous(h3))
            fontColour="#459977"
          >
            Premier League Predictor
          </glamorous(glamorous(h3))>
          <glamorous(p)>
            Play our awesome game
          </glamorous(p)>
          <glamorous(button)
            background="#459977"
            fullWidth={true}
            onClick={[Function]}
          >
            Play Now
          </glamorous(button)>
        </glamorous(div)>,
        "width": 400,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(glamorous(h3))
              fontColour="#459977"
            >
              Premier League Predictor
            </glamorous(glamorous(h3))>,
            <glamorous(p)>
              Play our awesome game
            </glamorous(p)>,
            <glamorous(button)
              background="#459977"
              fullWidth={true}
              onClick={[Function]}
            >
              Play Now
            </glamorous(button)>,
          ],
          "className": "ttt-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Premier League Predictor",
              "fontColour": "#459977",
            },
            "ref": null,
            "rendered": "Premier League Predictor",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Play our awesome game",
            },
            "ref": null,
            "rendered": "Play our awesome game",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "background": "#459977",
              "children": "Play Now",
              "fullWidth": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Play Now",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
